CMakeLists.txt

```diff
@@ -87,194 +87,89 @@ endfunction()
-ExternalProject_Add(g2tmpl
-  GIT_REPOSITORY "https://github.com/noaa-emc/nceplibs-g2tmpl"
-  GIT_TAG        "${tag}"
-  PREFIX         "${CMAKE_BINARY_DIR}/g2tmpl"
-  LIST_SEPARATOR |
-  CMAKE_ARGS     "-DCMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH_ALT_SEP}"
-                 "-DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}/g2tmpl/g2tmpl-${ver}"
-                 "-DENABLE_TESTS=${ENABLE_TESTS}"
-  )
+foreach(lib ${nceplibs})
+  getTagVer("${LIBCOMPS_FILE}" "${lib}" tag ver)
+  if(DEPLOY)
+    set(lib_install ${lib}/${lib}-${ver})
+  else()
+    set(lib_install ${lib}-${ver})
+  endif()
+  
+  ExternalProject_Add(${lib}
+    GIT_REPOSITORY "https://github.com/noaa-emc/nceplibs-${lib}"
+    GIT_TAG        "${tag}"
+    PREFIX         "${CMAKE_BINARY_DIR}/${lib}"
+    LIST_SEPARATOR |
+    CMAKE_ARGS     "-DCMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH_ALT_SEP}"
+    "-DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}/${lib_install}"
+    "-DENABLE_TESTS=${ENABLE_TESTS}"
+    DEPENDS ${${lib}_depends}
+    )
+endforeach()
 
-getTagVer("${LIBCOMPS_FILE}" "w3emc" tag ver)
-ExternalProject_Add(w3emc
-  GIT_REPOSITORY "https://github.com/noaa-emc/nceplibs-w3emc"
-  GIT_TAG        "${tag}"
-  PREFIX         "${CMAKE_BINARY_DIR}/w3emc"
-  LIST_SEPARATOR |
-  CMAKE_ARGS     "-DCMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH_ALT_SEP}"
-                 "-DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}/w3emc/w3emc-${ver}"
-                 "-DENABLE_TESTS=${ENABLE_TESTS}"
-  DEPENDS        nemsio sigio
-  )
 
 if(BUILD_CRTM)
   getTagVer("${LIBCOMPS_FILE}" "crtm" tag ver)
+  
+  if(DEPLOY)
+    set(lib_install crtm/crtm-${ver})
+  else()
+    set(lib_install crtm-${ver})
+  endif()
+
   ExternalProject_Add(crtm
     GIT_REPOSITORY "https://github.com/noaa-emc/emc_crtm"
     GIT_TAG        "${tag}"
     PREFIX         "${CMAKE_BINARY_DIR}/crtm"
     LIST_SEPARATOR |
     CMAKE_ARGS     "-DCMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH_ALT_SEP}"
-                   "-DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}/crtm/crtm-${ver}"
+                   "-DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}/${lib_install}"
     )
 endif()
 
 if(BUILD_POST)
   getTagVer("${LIBCOMPS_FILE}" "nceppost" tag ver)
+  
+  if(DEPLOY)
+    set(lib_install nceppost/nceppost-${ver})
+  else()
+    set(lib_install nceppost-${ver})
+  endif()
+  
   ExternalProject_Add(nceppost
     GIT_REPOSITORY "https://github.com/aerorahul/emc_post"
     GIT_TAG        "${tag}"
     GIT_SUBMODULES "CMakeModules"
     PREFIX         "${CMAKE_BINARY_DIR}/nceppost"
     LIST_SEPARATOR |
     CMAKE_ARGS     "-DCMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH_ALT_SEP}"
-                   "-DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}/nceppost/nceppost-${ver}"
-                   "-DOPENMP=${OPENMP}"
-                   "-DBUILD_POSTEXEC=OFF"
+    "-DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}/${lib_install}"
+    "-DOPENMP=${OPENMP}"
+    "-DBUILD_POSTEXEC=OFF"
     DEPENDS        bacio w3nco w3emc g2 g2tmpl ip sp crtm
     )
 endif()
```

Comment:
Can two functions after ``#Prepare to deploy modulefiles`` comment be moved inside ``if(DEPLOY)``?

Reply:
Fixed

Reply:
I think `DEPLOY` is the wrong option to key on.
We should be able to deploy regardless of install dir.
This has nothing to do with modulefiles, but everything to do with install directory structure.

Reply:
Are you saying we should have a different option solely for the install structure rather than using `DEPLOY`?

Reply:
yes! -DFLAT=OFF
Modulefiles are deployed or not based on -DDEPLOY, but they can be.
The FLAT option is for convenience, no reason why one cannot have modulefiles in the flat structure too.
The hierarchical structure caused issues that required the LIST_SEPARATOR fix.

Reply:
also, @DusanJovic-NOAA preparing the script does not necessarily mean it will be used.
So I suggest keep it outside of the if(DEPLOY). If a user wishes to change their mind after looking at it, they don't have to run cmake again.

Reply:
Ok. But then script and template files can be used (and prepared) only ``if (NOT FLAT)``

Reply:
what is the issue with preparing `deploy_modules.sh` script regardless of if(DEPLOY) or  if(FLAT)?
I can understand not wanting to add the target `make deploy` to `make all`, if one does not wish to deploy the modulefiles.

Reply:
modulefile templates assume hierarchical install tree ``${lib}/${lib}-${ver}``  ie. ``if (NOT FLAT)``.

Reply:
yes @DusanJovic-NOAA That will need to be updated as well.

Reply:
i.e. the templated modulefiles need to be consistent with FLAT or not FLAT.